src/gui/ui/imgui_gui.cpp

```diff
@@ -101,7 +99,8 @@ imgui_entry(World& world) {
         // TODO
         // https://stackoverflow.com/questions/23362497/how-can-i-resize-existing-texture-attachments-at-my-framebuffer
 
-        main_scene.update(window.value());
+        glfwGetWindowSize(window.value(), &window_width, &window_height);
```

Comment:
You should unpack `window` right after you check it.

Each call to `value` is a branch.